distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/PluginsManifest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.env.Environment;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This class models the contents of the {@code elasticsearch-plugins.yml} file. This file specifies all the plugins
+ * that ought to be installed in an Elasticsearch instance, and where to find them if they are not an official
+ * Elasticsearch plugin.
+ */
+public class PluginsManifest {
```

Comment:
Similar to my other comment, here we now call this a "manifest". We should converge on a name here. Is it "descriptor", or "manifest" or "config" or something else?

Reply:
I'm apt to call the file as a whole a plugins "config", and then we use "descriptor" only for individual entries in the "config". I lean towards "config" vs "manifest" because the former implies users should be editing this vs the latter is generally used to describe something that's immutable.

Reply:
I've renamed everything based on "config" instead of "manifest".